examples/gno.land/r/nt/boards2/v1/public.gno

```diff
@@ -310,6 +355,19 @@ func assertBoardExists(id BoardID) {
 	}
 }
 
+func assertBoardIsNotFrozen(b *Board) {
+	if b.IsReadOnly() {
+		panic("board is frozen")
+	}
+}
+
+func tryAssertBoardIsNotFrozen(bid BoardID) {
```

Comment:
imho `tryAssertBoardIsNotFrozen` is a bit confusing and it's not really distinguishable from func above.

wdyt about `mustGetWritableBoard`?

Also,  with that func all chunks with:

```go
board := mustGetBoard(bid)
assertBoardIsNotFrozen(board)
```

can be just replaced with:

```
board := mustGetWritableBoard(bid)
```

Reply:
The `tryAssertBoardIsNotFrozen` is definitely not an ideal name, you are right, but I couldn't think of a better one. For context, that "assert" function exists because there were three or four cases where that conditional code was duplicated. Maybe in this case is good to have that duplication 🤷 

The issue with the name is that in the context where the "assert" is used there might or might not be a board ID available, depending if the public realm function is operating on the realm permissions/DAO or a board one. That's why the assert starts with "try", the assertion would be tried only if the board ID is available.